From 5e42af52ade8644914ee0c5d9d6d44e53fceb3cc Mon Sep 17 00:00:00 2001
From: Michael Niedermayer <michael@niedermayer.cc>
Date: Mon, 12 Oct 2026 03:04:40 +0200
Subject: [PATCH] avcodec/aac/aacdec_usac: reject LPD channel streams

Found during triage/review of the security report GITTfi6VsK4u
(ANT-2026-5KX1BXM6), which does not report this.

(cherry picked from commit 5e42af52ade8644914ee0c5d9d6d44e53fceb3cc)
---
 libavcodec/aac/aacdec_usac.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/libavcodec/aac/aacdec_usac.c b/libavcodec/aac/aacdec_usac.c
index ae63761030d..a11e4b5ffcb 100644
--- a/libavcodec/aac/aacdec_usac.c
+++ b/libavcodec/aac/aacdec_usac.c
@@ -1788,7 +1788,8 @@ static int decode_usac_core_coder(AACDecContext *ac, AACUSACConfig *usac,
             ret = ff_aac_ldp_parse_channel_stream(ac, usac, ue, gb);
             if (ret < 0)
                 return ret;
-            continue;
+            avpriv_report_missing_feature(ac->avctx, "AAC USAC LPD");
+            return AVERROR_PATCHWELCOME;
         }
 
         if ((core_nb_channels == 1) ||
-- 
2.53.0

